rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictTypedCallRector/Fixture/same_typed_constant_int_returns.php.inc

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictTypedCallRector\Fixture;
+
+final class SameTypedIntConstantReturns
+{
+    const INT = 0;
+
+    public function getData()
+    {
+        if (rand(0,1)) {
+            return self::INT;
+        }
+
+        return $this->getInt();
+    }
+
+    public function getInt(): int
+    {
+
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictTypedCallRector\Fixture;
+
+final class SameTypedIntConstantReturns
+{
+    const INT = 0;
```

Comment:
how about add `@var string` here, is it still got `int` type?

Reply:
works like expected, I would say: https://github.com/rectorphp/rector-src/pull/4520/commits/e653c9ba817f837538cec348cceb0e951b07e215

(since constants cannot get the value changed at runtime)